[FIX] base_import: cannot import midnight date

ande-odoo opened the pull request:
Steps to reproduce:
---
1. Install hr_attendance
2. Go to attendances
3. Select any line
4. Change the check-out date to next day
5. Change the check-out time to 00:00:00
6. Export that line, file format xlsx
7. Favorites > import records
8. Upload the exported file > Test
9. Column check_out contains incorrect values.
Error in line 2: unconverted data remains: 2023-06-11

Cause of the issue:
---
options.get('date_format') is empty

opw-3374883

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/133424).

nle-odoo commented on addons/base_import/models/base_import.py:
```diff
@@ -1188,8 +1188,8 @@ def _parse_import_data_recursive(self, model, prefix, data, import_fields, optio
         fmt = fields.Date.to_string if field_type == 'date' else fields.Datetime.to_string
-        d_fmt = options.get('date_format')
-        dt_fmt = options.get('datetime_format')
+        d_fmt = options.get('date_format') or DEFAULT_SERVER_DATE_FORMAT
```
here since the issue seems to be only about datetime, I don't know if this part is necessary

nle-odoo reviewed (approved):
I can't reproduce, I guess because I don't have excel and libreoffice doesn't have this issue.

If I understand dbo, the issue happens when excel might remove the time part if the date 
 time is midnight, and somehow the `datetime_format` in this case is an empty string which doesn't work.

And I guess that the code for previewing probably uses the default format, so it would might make sense to have the default format here.

ande-odoo commented:
I updated the steps, the bug is now reproducible consistently.
It was the date_format that was empty, not datetime_format,
therefore I removed my fix for datetime_format (can be re-added if needed by the reviewer)

ryv-odoo commented:
Hello @ande-odoo , thank you.
It seems a good idea. IMO, you can fix datetime too, it should be easy to produce the same bug for it.

ryv-odoo commented:
Thank you @ande-odoo 
@robodoo r+
